datafusion/core/tests/tpcds_planning.rs

```diff
@@ -1052,9 +1052,12 @@ async fn regression_test(query_no: u8, create_physical: bool) -> Result<()> {
     for sql in &sql {
         let df = ctx.sql(sql).await?;
         let (state, plan) = df.into_parts();
-        let plan = state.optimize(&plan)?;
         if create_physical {
             let _ = state.create_physical_plan(&plan).await?;
```

Comment:
I still dont' get this part. Previously physical plan was created on top of optimized logical, why this step is skipped? 🤔

Reply:
`create_physical_plan` calls `optimize` itself already. Due to the fact that the logical plan was being optimised twice the logical vs physical schema mismatch was being hidden.

Reply:
To see what I'm talking about, comment out this line on `main` and run `tpcds_physical_q75`. You'll get
```
Error: Internal("Physical input schema should be the same as the one converted from logical input schema. Differences: \n\t- field nullability at index 5 [sales_cnt]: (physical) true vs (logical) false\n\t- field nullability at index 6 [sales_amt]: (physical) true vs (logical) false")
```